sdk/keyvault/azure-keyvault-administration/tests/test_examples_administration.py

```diff
@@ -0,0 +1,71 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import time
+
+from azure.core.credentials import AccessToken
+from azure.identity import DefaultAzureCredential
+from azure.keyvault.administration._internal import HttpChallengeCache
+from azure.keyvault.administration import KeyVaultBackupClient
+from devtools_testutils import ResourceGroupPreparer, StorageAccountPreparer
+import pytest
+from six.moves.urllib_parse import urlparse
+
+from _shared.helpers import mock
+from _shared.test_case import KeyVaultTestCase
+from blob_container_preparer import BlobContainerPreparer
+
+
+@pytest.mark.usefixtures("managed_hsm")
+class TestExamplesTests(KeyVaultTestCase):
+    def __init__(self, *args, **kwargs):
+        super(TestExamplesTests, self).__init__(*args, match_body=False, **kwargs)
+
+    def setUp(self, *args, **kwargs):
+        if self.is_live:
+            real = urlparse(self.managed_hsm["url"])
+            playback = urlparse(self.managed_hsm["playback_url"])
+            self.scrubber.register_name_pair(real.netloc, playback.netloc)
+        super(TestExamplesTests, self).setUp(*args, **kwargs)
+
+    def tearDown(self):
+        HttpChallengeCache.clear()
+        assert len(HttpChallengeCache._cache) == 0
+        super(KeyVaultTestCase, self).tearDown()
+
+    @property
+    def credential(self):
+        if self.is_live:
+            return DefaultAzureCredential()
+        return mock.Mock(get_token=lambda *_, **__: AccessToken("secret", time.time() + 3600))
+
+    @ResourceGroupPreparer(random_name_enabled=True, use_cache=True)
+    @StorageAccountPreparer(random_name_enabled=True)
+    @BlobContainerPreparer()
+    def test_example_backup_and_restore(self, container_uri, sas_token):
+        backup_client = KeyVaultBackupClient(self.managed_hsm["url"], self.credential)
+
+        # [START begin_backup]
+        # begin a vault backup
+        backup_poller = backup_client.begin_backup(container_uri, sas_token)
+
+        # check if the backup completed
+        done = backup_poller.done()
+
+        # get the final result
+        backup_operation = backup_poller.result()
```

Comment:
It might help to point out that `result()` blocks until the operation completes, and that the returned object gives a URL for the backup